### ISSUE TITLE
Is it possible to include all the prototypes of the JS Object?

### ISSUE THREAD

**Issue by alexshenker** [WebReflection/flatted#57]
Hi there,

The program works wonderfully.
In my project, I initialize an object that requires a lot of calculation. I am attempting to save this object
in JSON format so that I could use it later without recalculating.
The problem is that, once I parse it, it is missing all of its prototype methods (aside from normal obj and array methods). 
Could you advise on whether its possible to stringify the object with its prototypal chain?
Thank you!

**Comment by WebReflection** [comment]
The only thing this module does is to support recursion over **standard JSON behavior** and nothing else. so it's up to you find a way to provide enough details to stringify and revive an object with its prototype, not a goal of this project, not now, not ever, as the issue is beyond *JSON* capabilities, and everyone might have a different way to do so.

Things you need to consider to do so:

  * you need to somehow know which prototype needs to be stored, so that classes must be reachable via the reviver and its namespace, otherwise you can't possibly have back your instances
  * you need to know these classes through the namespace during stringify operation too
  * you need a specialized format that allows you to revive objects without ambiguity ... as example, by representing every object as `{"class": "Object", "json": {}}` for regular objects and `{"class":"YourThing",json:{}}` so that regular objects can be revived as Object class and others can be revived via `namespace.YourThing`

In short, all you want is to extend stringify and parse, but this has nothing to do with this module, it's something you need to do with *JSON* API (through this module, if you like, still just via JSON API).

**Comment by alexshenker** [comment]
> The only thing this module does is to support recursion over **standard JSON behavior** and nothing else. so it's up to you find a way to provide enough details to stringify and revive an object with its prototype, not a goal of this project, not now, not ever, as the issue is beyond _JSON_ capabilities, and everyone might have a different way to do so.
> 
> Things you need to consider to do so:
> 
> * you need to somehow know which prototype needs to be stored, so that classes must be reachable via the reviver and its namespace, otherwise you can't possibly have back your instances
> * you need to know these classes through the namespace during stringify operation too
> * you need a specialized format that allows you to revive objects without ambiguity ... as example, by representing every object as `{"class": "Object", "json": {}}` for regular objects and `{"class":"YourThing",json:{}}` so that regular objects can be revived as Object class and others can be revived via `namespace.YourThing`
> 
> In short, all you want is to extend stringify and parse, but this has nothing to do with this module, it's something you need to do with _JSON_ API (through this module, if you like, still just via JSON API).

Thank you, appreciate the advice!

**Comment by WebReflection** [comment]
@alexshenker I think I've just published what you are after https://github.com/WebReflection/json-instances#readme
